### ISSUE TITLE
`if-with-same-arms (SIM114)` false-negative with comments

### ISSUE THREAD

**Issue by Avasam** [astral-sh/ruff#23080]
### Summary

[if-with-same-arms (SIM114)](https://docs.astral.sh/ruff/rules/if-with-same-arms/#if-with-same-arms-sim114) does not trigger for a branch where the only difference is a comment:

```py
while True:
    rc = win32event.MsgWaitForMultipleObjects(
        (StopEvent,), 0, TIMEOUT, win32event.QS_ALLEVENTS
    )

    if rc == win32event.WAIT_OBJECT_0:
        break
    elif rc == win32event.WAIT_OBJECT_0 + 1 and pythoncom.PumpWaitingMessages():
        break  # wm_quit
```
(remove `vm_quit` and it'll trigger)

This should of course count as an unsafe-fix (https://github.com/astral-sh/ruff/issues/19576), but should still trigger the rule.

### Version

0.11 (but I also tried with `uvx ruff@0.15 check --select=SIM114`

**Comment by ntBre** [comment]
I'd have to read back through my comments on #22079 to be sure, but I _think_ this is an intentional negative for the rule. We take the presence of a comment as a signal that the author wanted the blocks to be separate. There are some tricky cases with end-of-line noqa comments, but I think this falls under the intentional heuristic.

**Comment by Avasam** [comment]
I can see the argument for the heuristic. If it's intentional then it's not a bug. Especially if intended to reduce noise.

I think the comment can (manually) be moved inside the condition.

But I won't fight for it, I really opened this issue thinking I found a bug. I'm fine if you close with "working as intended".
